### DISCUSSION TITLE
Get all opc nodes from one server

### DISCUSSION THREAD

**Discussion by Yoshibeatsu** [FreeOpcUa/opcua-asyncio#1653] [Q&A]
Hi, it might be not common practice but i need to get all the opc nodes from one server. I couldn't find an example which is getting all the nodes and also is using certificates and user login as well. I'm quite new to programming especially python. The nodes are in folders and sub sub folders. So I think i need to make a recursive search ...

I tried already to develop something in C# with the libary by the opc foundation. It worked on my develop system but ran into a timeout after a while on a real opc server....

Thank you guys in advance

**Comment by prs4** [comment]
Hi, You are right about the recursive search - I use a simple script based on **read_browse_name()** and **get_children()** methods from asyncua to gather all nodes from a PLC. However I suppose that the timeout problem points to some connection issues which you have to solve first.
